src/plugins/intel_gpu/src/graph/impls/ocl/crop.cpp

```diff
@@ -26,19 +26,62 @@ struct crop_impl : typed_primitive_impl_ocl<crop> {
     }
 
 public:
-    static kernel_params_t get_kernel_params(const kernel_impl_params& impl_param) {
-        auto params = get_default_params<kernel_selector::eltwise_params>(impl_param);
+    static kernel_params_t get_kernel_params(const kernel_impl_params& impl_param, bool is_shape_agnostic = false) {
+        const auto& primitive = impl_param.typed_desc<crop>();
+        auto params = get_default_params<kernel_selector::eltwise_params>(impl_param, is_shape_agnostic);
         auto optional_params = get_default_optional_params<kernel_selector::eltwise_optional_params>(impl_param.get_program());
 
         params.operations.push_back({{kernel_selector::eltwise_params::InputType::Buffer(0)}, kernel_selector::eltwise_mode::ASSIGN});
-        params.inputs[0] = convert_data_tensor(impl_param.get_input_layout(), impl_param.input_offsets[0]);
+        if (impl_param.get_program().get_node(primitive->id).is_dynamic()) {
+            // WA to always match compiled dynamic kernel with dispatch data
+            // W/O enforcing this option we may generate kernel for "broadcast" scneario due to umatched tensor dimensions
+            // but in runtime dispatch data will be generated for non-broadcast case as shapes are actually same.
+            params.broadcast = true;
+        } else {
+            params.inputs[0] = convert_data_tensor(impl_param.get_input_layout(), impl_param.input_offsets[0]);
+        }
         return {params, optional_params};
     }
+        void update_dispatch_data(const kernel_impl_params& impl_param) override {
+            auto kernel_params = get_kernel_params(impl_param, true);
+            auto runtime_offset = convert_data_tensor(impl_param.get_input_layout(), impl_param.input_offsets[0]).GetFirstElementOffset();
+            kernel_selector::ScalarDescriptor s;
+            s.t = kernel_selector::ScalarDescriptor::Types::UINT32;
+            s.v.u32 = runtime_offset;
+
+            if (_kernel_data.kernels[0].params.scalars.empty())
```

Comment:
Shouldn't we throw an exception in that case? Because proper kernel already should be configured to have scalars, isn't it?

Reply:
In shape agnostic kernel's creation I didnt added initial param scalar value. 
I intended to add that value at the initial run of the shape agnostic kernel. 
However we can change the behavior as well, 
- set at creation => use it

Reply:
Applied.